### PR TITLE
Fix the problem that alter modfiy order by causes inconsistent metadata

### PR DESCRIPTION
**Pull request by iceFireser**
Modify order by can cause replicated database tables to differ.
There are three ClickHouse servers here:
shard1、repilica1
shard2、repilica1
shard2、repilica2

First, execute SQL on shard1, replica1:  
CREATE DATABASE alter_modify_order_by ENGINE = Replicated('/test/database/alter_modify_order_by', 'shard2', 'replica1');
CREATE TABLE alter_modify_order_by.t1 (id Int64, score Int64) ENGINE = ReplicatedMergeTree('/test/tables/{uuid}/{shard}', '{replica}') ORDER BY (id);
ALTER TABLE alter_modify_order_by.t1 modify order by (id);

Then,  execute SQL on shard2, replica1:  
CREATE DATABASE alter_modify_order_by ENGINE = Replicated('/test/database/alter_modify_order_by', 'shard2', 'replica1');

Then,  execute SQL on shard1, replica1: 
show create table alter_modify_order_by.t1;
result:  
CREATE TABLE alter_modify_order_by.t1
(
    `id` Int64,
    `score` Int64
)
ENGINE = ReplicatedMergeTree('/test/tables/{uuid}/{shard}', '{replica}')
ORDER BY id
SETTINGS index_granularity = 8192

Then,  execute SQL on shard2, replica1: 
show create table alter_modify_order_by.t1;
result:  
CREATE TABLE alter_modify_order_by.t1
(
    `id` Int64,
    `score` Int64
)
ENGINE = ReplicatedMergeTree('/test/tables/{uuid}/{shard}', '{replica}')
PRIMARY KEY id
ORDER BY id
SETTINGS index_granularity = 8192

 **Inconsistent metadata**

Then,  execute SQL on shard2, replica1: 
CREATE TABLE alter_modify_order_by.t2 (id Int64, score Int64) ENGINE = ReplicatedMergeTree('/test/tables/{uuid}/{shard}', '{replica}') ORDER BY (id);
ALTER TABLE alter_modify_order_by.t2 modify order by (id);

Then,  execute SQL on shard2, replica2: 
CREATE DATABASE alter_modify_order_by ENGINE = Replicated('/test/database/alter_modify_order_by', 'shard2', 'replica2');

ClickHouse encountered an error, as shown in the logs：<Error> DDLWorker(alter_modify_order_by): Error on initialization of alter_modify_order_by: Code: 342. DB::Exception: Existing table metadata in ZooKeeper differs in sorting key expression. Stored in ZooKeeper: , parsed from ZooKeeper: , local: id. (METADATA_MISMATCH)

### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix the problem that alter modfiy order by causes inconsistent metadata

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: src/Storages/AlterCommands.cpp [hunk 1]
```diff
@@ -1245,6 +1245,13 @@ void AlterCommands::prepare(const StorageInMemoryMetadata & metadata)
 {
     auto columns = metadata.columns;
 
+    auto ast_to_str = [](const ASTPtr & query) -> String
+    {
+        if (!query)
+            return "";
+        return queryToString(query);
+    };
```
Comment (yariks5s) [inline src/Storages/AlterCommands.cpp:1253]: Why not change the `queryToString` implementation to return an empty string if there is no `query` argument?

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 16d7ff11be71df496a960ba833649df2a452a24b with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/67436/16d7ff11be71df496a960ba833649df2a452a24b/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/67436/16d7ff11be71df496a960ba833649df2a452a24b/style_check.html">❌ failure</a></td></tr>
<tbody>
</table>

File: src/Storages/AlterCommands.cpp (hunk 1 above)
Comment (iceFireser) [reply src/Storages/AlterCommands.cpp:1253]: queryToString is a public function that is called in many places. I'm concerned that modifying queryToString might affect a lot of logic.

**Comment by yariks5s** [comment]
hi, freebsd build is fixed on master, can you merge please?

**Comment by iceFireser** [comment]
> Just pushing empty commits won't help, you need to sync your branch with ClickHouse:master: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork

Thank you. I will merge.

**Review by yariks5s — APPROVED** [review]

**Pull request merged**